addons/common/fnc_addMagazine.sqf

```diff
@@ -8,50 +8,75 @@ Description:
     case of trouble, or true when able to add _item to _unit.
 
 Parameters:
-    _unit - the unit or vehicle
-    _item - name of the magazine to add
+    _unit   - the unit or vehicle <OBJECT>
+    _item   - name of the magazine to add <STRING>
+    _ammo   - ammo count <NUMBER>
+    _verify - if true, then put item on the ground if it can't be added <BOOLEAN>
 
 Returns:
-    true on success, false otherwise
+    true on success, false otherwise <BOOLEAN>
 
 Examples:
     (begin example)
-    _result = [player, "SmokeShell"] call CBA_fnc_addMagazine
+        _result = [player, "SmokeShell"] call CBA_fnc_addMagazine
     (end)
 
 Author:
 
 ---------------------------------------------------------------------------- */
-
 #include "script_component.hpp"
 SCRIPT(addMagazine);
 
-#define __scriptname fAddMagazine
+params [["_unit", objNull, [objNull]], ["_item", "", [""]], ["_ammo", -1, [0]], ["_verify", false, [false]]];
 
-#define __cfg (configFile >> "CfgMagazines")
-#define __action addMagazine
+private _return = false;
 
-params ["_unit","_item"];
-if (typeName _unit != "OBJECT") exitWith {
-    TRACE_2("Unit not Object",_unit,_item);
-    false
-};
-if (typeName _item != "STRING") exitWith {
-    TRACE_2("Item not String",_unit,_item);
-    false
-};
 if (isNull _unit) exitWith {
-    TRACE_2("Unit isNull",_unit,_item);
-    false
+    TRACE_2("Unit not Object or null",_unit,_item);
+    _return
 };
-if (_item == "") exitWith {
-    TRACE_2("Empty Item",_unit,_item);
-    false
+
+if (_item isEqualTo "") exitWith {
+    TRACE_2("Item not String or empty",_unit,_item);
+    _return
+};
+
+private _config = configFile >> "CfgMagazines" >> _item;
+
+if (!isClass _config || {getNumber (_config >> "scope") < 2}) exitWith {
+    TRACE_2("Item does not exist in Config",_unit,_item);
+    _return
 };
-if !(isClass (__cfg >> _item)) exitWith {
-    TRACE_2("Item not exist in Config",_unit,_item);
-    false
+
+if (_verify) then {
+    if (_unit canAdd _item) then {
+        if (_ammo < 0) then {
+            _unit addMagazine [_item, 1E6]; // addMagazine STRING is not synched when used on remote units. addMagazine ARRAY is.

```

Comment:
It's not a million magazines, but one magazine with full ammo.
See: https://community.bistudio.com/wiki/addMagazine_array
`addMagazine STRING` fails on remote units. The alternate syntax `addMagazine ARRAY` does work on remote units.
I've tested this in local hosted MP and it's true. `addMagazine` does desynch the inventory causing weird issues when used on remote units. This is part of the fix to get this function working on all units regardless of locality.